### PR TITLE
Azure branch

### PR DESCRIPTION
**Pull request by mheine3000**
PR to open source the work done on the (incomplete) Azure pipeline.

### CODE REVIEW

File: azure/pipeline/azure_tools.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import planetary_computer
+
+def get_fs(account='nrel', container='oedi'):
+    return planetary_computer.get_adlfs_filesystem(account, container)
+
+def get_size(path, units='B'):
+    fs = get_fs()
+    size = fs.du(path, total=True)
+    match units:
```
Comment (jgu2) [inline azure/pipeline/azure_tools.py:9]: This syntax requires Python version >=3.10, 

We need to upgrade our CI testing Python version, it's 3.9, which raised syntax error
```bash
./azure/pipeline/azure_tools.py:9:11: E999 SyntaxError: invalid syntax
    match units:
          ^
1     E999 SyntaxError: invalid syntax
1
Error: Process completed with exit code 1.
```

File: azure/pipeline/update_trans_container.sh [hunk 2]
```diff
@@ -0,0 +1,4 @@
+docker build -t transform_h5_container -f ./pipeline/transform_h5_container/Dockerfile .
+docker tag transform_h5_container:latest 351672045885.dkr.ecr.us-west-2.amazonaws.com/transform_h5_container:latest
+aws ecr get-login-password --region us-west-2 | docker login --username AWS --password-stdin 351672045885.dkr.ecr.us-west-2.amazonaws.com
+docker push 351672045885.dkr.ecr.us-west-2.amazonaws.com/transform_h5_container:latest
```
Comment (jgu2) [inline azure/pipeline/update_trans_container.sh:4]: The docker image released to dockerhub does not relay on `azure` module, right? If not, then can exclude `azure` directory in `.dockerignore`, so to keep docker image size small.

File: LICENSE [hunk 3]
```diff
@@ -1,6 +1,6 @@
 BSD 3-Clause License
 
-Copyright (c) 2022 Alliance for Sustainable Energy, LLC
+Copyright (c) 2023 Alliance for Sustainable Energy, LLC and Skye Analytics, Inc.
```
[the remark this answers is not in the document]
Comment (mheine3000) [reply LICENSE:3]: Done

File: azure/az_cli_guide.txt [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+az storage blob list --account-name nrel --container-name nrel-nsrdb --output table --sas-token "st=2023-05-22T16%3A44%3A49Z&se=2023-05-23T17%3A29%3A50Z&sp=rl&sv=2021-06-08&sr=c&skoid=c85c15d6-d1ae-42d4-af60-e2ca0f81359b&sktid=72f988bf-86f1-41af-91ab-2d7cd011db47&skt=2023-05-23T16%3A44%3A48Z&ske=2023-05-30T16%3A44%3A48Z&sks=b&skv=2021-06-08&sig=Sip4D9k3rFo/SeCcGxp5I11X6hL/H2AWt9bREQWZAlY%3D"
+
+
+
+
+az storage blob list --account-name nrel --container-name oedi --prefix PR100 --sas-token 'sv=2020-08-04&si=oedi-ro&sr=c&sig=O%2BQvKRV9uYuK36WzVRoCJdFO%2BRifXO8aIGqbS%2F3llPs%3D'
+
+az storage blob list --output table --account-name nrel --container-name nrel-nsrdb --sas-token 'sv=2020-08-04&si=nrel-nsrdb-ro&sr=c&sig=H8GUesZmOXzMomMWdrnXQv2ZPI09hANqHIcVrP7Ejl0%3D'
+
+az storage blob list --account-name nrel --container-name oedi --prefix PR100 --sas-token 'sv=2020-08-04&si=nrel-nsrdb-ro&sr=c&sig=H8GUesZmOXzMomMWdrnXQv2ZPI09hANqHIcVrP7Ejl0%3D'
```
[the remark this answers is not in the document]
Comment (mheine3000) [reply azure/az_cli_guide.txt:42]: Done

File: azure/pipeline/azure_tools.py (hunk 1 above)
Comment (mheine3000) [reply azure/pipeline/azure_tools.py:9]: changed to if structure

File: azure/pipeline/update_trans_container.sh (hunk 2 above)
Comment (mheine3000) [reply azure/pipeline/update_trans_container.sh:4]: The Docker image does require some files from the azure module. If you open the Dockerfile, you will see that I only copy exactly what I need, so the image is already as small as possible.

I don't think an update to .dockerignore is needed, but let me know if I'm misunderstanding something!

File: azure/pipeline/aws_tools.py [hunk 5]
```diff
@@ -0,0 +1,360 @@
...
+    #         if resolution == 'hourly':
+    #             files = s3.glob(f'{bucket}/{prefix}/hourly/*.h5')
+    #         elif resolution == '5min':
+    #             files = s3.glob(f'{bucket}/{prefix}/5min/*.h5')
+    #     else:
+    #         if resolution == 'hourly':
+    #             files = s3.glob(f'{bucket}/{prefix}/*.h5')
+    #         elif resolution == '5min':
+    #             files = s3.glob(f'{bucket}/{prefix}/*/*.h5')
+    # return files
```
[the remark this answers is not in the document]
Comment (mheine3000) [reply azure/pipeline/aws_tools.py:89]: Cleaned up

**Review by jgu2 — APPROVED** [review]

**Pull request merged**